### DISCUSSION TITLE
Unable to see logs when running pytest

### DISCUSSION THREAD

**Discussion by astrochun** [Textualize/rich#2256] [Q&A]
I have a basic logging configuration as follow:

```python
import logging

from rich.logging import RichHandler

FORMAT = "%(message)s"

logging.basicConfig(
    level=logging.INFO,
    format=FORMAT,
    datefmt="[%X]",
    handlers=[RichHandler(markup=True, show_path=False)],
)

app_log = logging.getLogger("rich")
```

I've configured `pytest` and `coverage` as follow in my pyproject.toml:
```ini
[tool.pytest.ini_options]
addopts = "-sv --asyncio-mode=strict"
testpaths = [
    "tests",
]

[tool.coverage.paths]
source = ["src"]

[tool.coverage.run]
branch = true
command_line = "-m pytest"
source = ["src"]

[tool.coverage.report]
show_missing = true
precision = 2
skip_empty = true
```

When I run a basic unit test of multiple `app_log.*()` calls, I get the following:
```
tests/test_config.py::test_config test
PASSED
tests/test_logger.py::test_logger PASSED
```

I would have expected to see the log calls but it seems that with a `basicConfig` it doesn't work. It works via `dictConfig` in another project. I was hoping to use `basicConfig` for simplicity.

**Comment by astrochun** [comment]
Actually, I think I came up with a fix. See: https://docs.pytest.org/en/7.1.x/reference/reference.html#confval-log_cli

What I don't understand is why this works; however, in another repo where is use `logging.dictConfig` it still prints the log even without setting this as `true`.

**Reply by willmcgugan** [reply]
Honestly, don't know. The RichHandler just writes to stdout. I wouldn't be surprised if pytest does some magic to capture logs, but isn't aware of the RichHandler.
